front50-core/src/main/groovy/com/netflix/spinnaker/front50/listeners/ChaosMonkeyEventListener.java

```diff
@@ -64,47 +59,28 @@ public boolean supports(ApplicationEventListener.Type type) {
 
   @Override
   public Application call(Application originalApplication, Application updatedApplication) {
-    boolean isChaosMonkeyEnabled = isChaosMonkeyEnabled(updatedApplication);
-    if (isChaosMonkeyEnabled(originalApplication) == isChaosMonkeyEnabled) {
-      // Flag didn't change, we don't need to do anything.
-      return updatedApplication;
-    }
+    Application.Permission permission =
+        applicationPermissionsService.getApplicationPermission(updatedApplication.getName());
 
-    Object rawPermissions = updatedApplication.details().get("permissions");
-    if (!(rawPermissions instanceof Map)) {
-      // Exit early, no permissions config found on the application
-      log.warn("No permissions config found on application '{}'", updatedApplication.getName());
+    if (!permission.getPermissions().isRestricted()) {
```

Comment:
Will this cover the case where my application was restricted and had chaos enabled and I now want to remove the restriction? Basically going from `me` to `` (from the UI's perspective).

If so, perhaps a test permutation for that would be useful?

Reply:
Good catch.  I added a check for this, including some tests.